amd64/root/firmware-update.sh

```diff
@@ -156,11 +160,13 @@ if [[ ! -f ${FLASHER_PARAM_VALUES[-f]} ]]; then
     echo " "
     echo "Downloading..."
     echo " "
-    curl --fail --output "${FLASHER_PARAM_VALUES[-f]}" "${FW_ONLINE_BASE%/}/$fileName" && [[ -f ${FLASHER_PARAM_VALUES[-f]} ]]
-    delete_and_exit_on_error "${FLASHER_PARAM_VALUES[-f]}" "Download Error! Please re-run this script..."
+    curl --fail --silent --output "${FLASHER_PARAM_VALUES[-f]}" "${FW_ONLINE_STABLE_BASE%/}/$fileName" || \
+    curl --fail --silent --output "${FLASHER_PARAM_VALUES[-f]}" "${FW_ONLINE_BETA_BASE%/}/$fileName"
```

Comment:
Instead of hardcoding the two targets I would prefer a solution which processes a list of possible urls. In that case you will know which one succeeded ...
But I think this will make `delete_and_exit_on_error` unusable here.